src/transformers/modeling_tf_utils.py

```diff
@@ -104,6 +106,41 @@ def get_config(self):
     return cls
 
 
+class TFQuestionAnsweringLoss:
+    def compute_loss(self, labels, logits):
+        loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(
+            from_logits=True, reduction=tf.keras.losses.Reduction.NONE
+        )
+        start_loss = loss_fn(labels["start_position"], logits[0])
+        end_loss = loss_fn(labels["end_position"], logits[1])
+
+        return (start_loss + end_loss) / 2.0
+
+
+class TFTokenClassificationLoss:
+    def compute_loss(self, labels, logits):
+        loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(
+            from_logits=True, reduction=tf.keras.losses.Reduction.NONE
+        )
+        active_loss = tf.reshape(labels, (-1,)) != -1
+        reduced_logits = tf.boolean_mask(tf.reshape(logits[0], (-1, shape_list(logits[0])[2])), active_loss)
+        labels = tf.boolean_mask(tf.reshape(labels, (-1,)), active_loss)
+
+        return loss_fn(labels, reduced_logits)
+
+
+class TFSequenceClassificationAndMultipleChoiceLoss:
```

Comment:
should we split into two different classes?

Reply:
My thought was, as it is the exact same loss computation why not merge the two names, but your proposal might be more insightful indeed.

Reply:
Maybe just alias one to the other or do a trivial sub-class

Reply:
Very good point, I will do the update

Reply:
Should be ok now.